apps/examples/testcase/ta_tc/wifi_manager/itc/itc_wifi_manager_main.c

```diff
@@ -0,0 +1,358 @@
+#include <pthread.h>
+#include <sys/time.h>
+#include <errno.h>
+#include <wifi_manager/wifi_manager.h>
+#include "tc_common.h"
+
+extern sem_t tc_sem;
+extern int working_tc;
+
+static pthread_mutex_t g_wifi_manager_test_mutex = PTHREAD_MUTEX_INITIALIZER;
+static pthread_cond_t g_wifi_manager_test_cond;
+
+#define WIFITEST_SIGNAL\
+    do {\
+        pthread_mutex_lock(&g_wifi_manager_test_mutex);\
+        pthread_cond_signal(&g_wifi_manager_test_cond);         \
+        pthread_mutex_unlock(&g_wifi_manager_test_mutex);       \
+} while (0)
+
+#define WIFITEST_WAIT\
+    do {\
+        pthread_mutex_lock(&g_wifi_manager_test_mutex);         \
+        pthread_cond_wait(&g_wifi_manager_test_cond, &g_wifi_manager_test_mutex);   \
+        pthread_mutex_unlock(&g_wifi_manager_test_mutex);       \
+    } while (0)
+
+static void wifi_sta_connected_cb(void);       // in station mode, connected to ap
+static void wifi_sta_disconnected_cb(void);    // in station mode, disconnected from ap
+static void wifi_scan_ap_done_cb(wifi_manager_scan_info_s **scan_info, wifi_manager_scan_result_e res); // called when scanning ap is done
+
+static wifi_manager_cb_s wifi_callbacks = {
+    wifi_sta_connected_cb,
+    wifi_sta_disconnected_cb,
+    NULL,
+    NULL,
+    wifi_scan_ap_done_cb,  // this callback function is called when scanning ap is done.
+};
+
+static void wifi_sta_connected_cb(void)
+{
+    printf("wifi_sta_connected: send signal!!! \n");
+    WIFITEST_SIGNAL;
+}
+
+static void wifi_sta_disconnected_cb(void)
+{
+    printf("wifi_sta_disconnected: send signal!!! \n");
+    WIFITEST_SIGNAL;
+}
+
+static void wifi_scan_ap_done_cb(wifi_manager_scan_info_s **scan_info, wifi_manager_scan_result_e res)
+{
+    /* Make sure you copy the scan results onto a local data structure.
+     * * It will be deleted soon eventually as you exit this function.
+     * */
+    if (res == WIFI_SCAN_FAIL) {
+        printf("WiFi scan failed\n");
+        return;
+    }
+    wifi_manager_scan_info_s *wifi_scan_iter = *scan_info;
+    while (wifi_scan_iter != NULL) {
+        printf("SSID: %-20s, BSSID: %-20s, RSSI: %d, CH: %d, Phy_type: %d\n", \
+                wifi_scan_iter->ssid, wifi_scan_iter->bssid, wifi_scan_iter->rssi, \
+                wifi_scan_iter->channel, wifi_scan_iter->phy_mode);
+        wifi_scan_iter = wifi_scan_iter->next;
+    }
+}
+
+static int wifi_test_signal_init(void)
+{
+    int res = pthread_mutex_init(&g_wifi_manager_test_mutex, NULL);
+    if (res != 0) {
+        printf("wifi test signal init fail(%d) (%d)\n", res, errno);
+        return -1;
+    }
+    res = pthread_cond_init(&g_wifi_manager_test_cond, NULL);
+    if (res != 0) {
+        printf("wifi test condition init fail(%d) (%d)\n", res, errno);
+        res = pthread_mutex_destroy(&g_wifi_manager_test_mutex);
+        if (res != 0) {
+            printf("wifi test signal init fail(%d) (%d)\n", res, errno);
+            return -1;
+        }
+        return -1;

```

Comment:
In both` res == 0` and` res != 0 `conditions it's returning `-1` , I think `return -1;` at `L#102` can be removed.
or else anyway as you are in failed case, I think no need to check again for fail case at` L#100`